arcgis-ios-sdk-samples/Display information/Create symbol styles from web styles/CreateSymbolStylesFromWebStylesViewController.swift

```diff
@@ -0,0 +1,308 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class CreateSymbolStylesFromWebStylesViewController: UIViewController, UIAdaptivePresentationControllerDelegate {
+    // MARK: Storyboard views
+    
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap(featureLayer: featureLayer)
+            mapView.setViewpoint(
+                AGSViewpoint(
+                    latitude: 34.28301,
+                    longitude: -118.44186,
+                    scale: 1e4
+                )
+            )
+        }
+    }
+    /// The button to show legend table.
+    @IBOutlet var legendBarButtonItem: UIBarButtonItem!
+    
+    // MARK: Instance properties
+    
+    /// The observation on the map view's `mapScale` property.
+    var mapScaleObservation: NSKeyValueObservation?
+    /// The Esri 2D point symbol style created from a web style.
+    let symbolStyle = AGSSymbolStyle(styleName: "Esri2DPointSymbolsStyle", portal: .arcGISOnline(withLoginRequired: false))
+    
+    private var symbolsDataSource: SymbolsDataSource?
+    private var symbols = [(category: SymbolCategory, symbol: AGSSymbol)]() {
+        didSet {
+            symbolsDataSource?.symbols = symbols
```

Comment:
I'm trying to address the second comment here: https://github.com/Esri/arcgis-runtime-samples-ios/pull/1038#discussion_r583294741 about detecting potential changes to the symbols.

Initially I thought it is as easy as passing the array as a reference. But there are a few impediments. Therefore I came up with this rather not ideal solution. Please let me know the better way to do this!